src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ActiveDirectorySite.cs

```diff
@@ -1302,16 +1302,18 @@ private void GetDomains()
 
                 IntPtr info = (IntPtr)0;
                 // call DsReplicaSyncAllW
-                IntPtr functionPtr = global::Interop.Kernel32.GetProcAddress(DirectoryContext.ADHandle, "DsListDomainsInSiteW");
-                if (functionPtr == (IntPtr)0)
+                var dsListDomainsInSiteW = (delegate* unmanaged<IntPtr, char*, IntPtr*, int>)global::Interop.Kernel32.GetProcAddress(DirectoryContext.ADHandle, "DsListDomainsInSiteW");
```

Comment:
Is it desirable that this is `IntPtr*` rather than `DS_NAME_RESULT*`?

What about `int` rather than `uint` (since its a `DWORD`)?

Reply:
I wanted to match the existing delegate signatures to ensure I wouldn't have to add a ton of casts.